scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -1261,12 +1234,16 @@ def verify_platforms_existence(self, platform_names_to_verify, log_info=""):
         as platform_allow or integration_platforms options) is correct. If not -
         log and raise error.
         """
+        _platforms = []
         for platform in platform_names_to_verify:
             if platform in self.platform_names:
-                continue
+                p = self.get_platform(platform)
+                if p:
+                    _platforms.append(p.name)
             else:
                 logger.error(f"{log_info} - unrecognized platform - {platform}")
                 sys.exit(2)
+        return _platforms
```

Comment:
with this additional behavior the result needs to be used at apply_filters(), from line 901:
```
                platforms_ = self.verify_platforms_existence(ts.platform_allow, f"{ts_name} - platform_allow")
                a = set(platform_scope)
                b = set(filter(lambda item: item.name in platforms_, self.platforms))
                c = a.intersection(b)
                if not c:
                    platform_scope = list(
                        filter(lambda item: item.name in platforms_, self.platforms)
                    )
```

Reply:
this call needs to be removed, we do this verification already, so this is
another duplicate, looks like I missed it.